### PR TITLE
Implemented position ordering

### PR DESCRIPTION
**Pull request by peiva-git**
Ordering based on chess board cells' coordinates, see code comments and proposed documentation

### CODE REVIEW

File: src/test/java/PositionTests.java [hunk 1]
```diff
@@ -48,4 +59,17 @@ void testNewPosition(int row, int column, Class<Exception> expectedException) {
...
+
+    @ParameterizedTest
+    @MethodSource("providePositionPairsForComparison")
+    void testPositionOrdering(Position first, Position second, int expectedResult) {
+        if (expectedResult == 1) {
+            assertTrue(first.compareTo(second) > 0);
+        } else if (expectedResult == -1) {
+            assertTrue(first.compareTo(second) < 0);
+        } else {
+            assertEquals(0, first.compareTo(second));
```
Comment (manuelkoso) [inline src/test/java/PositionTests.java:71]: I would do `assertTrue(first.compareTo(second) == 0)` in order to have the same structure for all the three cases. You can laso add `assertNotEquals(first, second)` for the first and the second case

File: src/main/java/Position.java [hunk 2]
```diff
@@ -67,4 +67,33 @@ private boolean isPositionOneColumnToTheLeft(@NotNull Position otherPosition) {
...
+     * 2 -  -  -  -  -  -  -  -
+     * 1 -  -  -  -  -  -  -  -
+     *   A  B  C  D  E  F  G  H
+     * **White player's side is here**
+     */
+    @Override
+    public int compareTo(@NotNull Position position) {
+        if (column == position.getColumn()) {
+            return position.getRow() - row;
+        } else return Integer.compare(position.getColumn(), column);
```
Comment (manuelkoso) [inline src/main/java/Position.java:97]:
```java
return position.getColumn() - column;
```
and you can remove the last `else`

File: src/test/java/PositionTests.java (hunk 1 above)
Comment (peiva-git) [reply src/test/java/PositionTests.java:71]: Correction suggested by IntelliJ, I don't know what's the reason for it

File: src/main/java/Position.java [hunk 3]
```diff
@@ -67,4 +67,34 @@ private boolean isPositionOneColumnToTheLeft(@NotNull Position otherPosition) {
...
+     * 1 -  -  -  -  -  -  -  -
+     *   A  B  C  D  E  F  G  H
+     * **White player's side is here**
+     */
+    @Override
+    public int compareTo(@NotNull Position position) {
+        if (column == position.getColumn()) {
+            return position.getRow() - row;
+        }
+        return Integer.compare(position.getColumn(), column);
```
[the remark this answers is not in the document]
Comment (manuelkoso) [reply src/main/java/Position.java:98]: If you want you can use the `Integer.compare()` also for the above statement:
```java
return Integer.compare(position.getRow(), row);
```

[the remark this answers is not in the document]
Comment (peiva-git) [reply src/main/java/Position.java:98]: The suggested code is equivalent to this:
```java
        if (column < position.getColumn()) {
            return 1;
        } else if (column > position.getColumn()) {
            return -1;
        } else {
            return position.getRow() - row;
        }
```
Maybe this is clearer?

[the remark this answers is not in the document]
Comment (manuelkoso) [reply src/main/java/Position.java]: I think that this is clearer:
```java
    @Override
    public int compareTo(@NotNull Position position) {
        if (column == position.getColumn()) {
            return Integer.compare(position.getRow(), row);
        }
        return Integer.compare(position.getColumn(), column);
    }
```

[the remark this answers is not in the document]
Comment (manuelkoso) [reply src/main/java/Position.java]: But we can leave your solution, it's no big deal

**Review by manuelkoso — APPROVED** [review]
See my comment

**Comment by manuelkoso** [comment]
Well done Ivan, I am proud of you

**Pull request merged**